### PR TITLE
Store local energy deposition

### PR DESCRIPTION
**Pull request by amandalund**
Small addition based on our discussion at today’s physics meeting and on #89: this lets us to keep track of local energy deposition in an interaction.

### CODE REVIEW

File: test/physics/em/PhotoelectricInteractor.test.cc
Comment (sethrj) [inline test/physics/em/PhotoelectricInteractor.test.cc:117]: @vrpascuzzi Did you end up separating the momentum and energy conservation checks? We could add that to this PR.

**Review by sethrj — COMMENTED** [review]
Thanks!

Let's wait to hear back from Vincent on the momentum/energy test before merging. Either way, I think it would also be good to add `energy_deposition` "gold" values in your livermore test as well just as a quick check that it's being initialized and that the results are sensible.

File: test/physics/em/PhotoelectricInteractor.test.cc
Comment (vrpascuzzi) [reply test/physics/em/PhotoelectricInteractor.test.cc:117]: @sethrj,

You mentioned doing this in such a way that we wouldn't need to make modifications to existing tests that call `check_conservation`. My initial thought was to change the function definition:
```
void check_conservation(const Interaction& interaction) const;
```
to
```
void check_conservation(const Interaction& interaction, bool do_energy, bool do_momentum) const;
```
But maybe we could include conservation information directly into the `Interaction` class?

Comment (sethrj) [reply test/physics/em/PhotoelectricInteractor.test.cc:117]: OK, my thought was initially we split up the existing `check_conservation` into two functions, and leave the original  signature for compatibility and simplicity:
```c++
void check_conservation(const Interaction& interaction) const
{
    this->check_energy_conservation(interaction);
    this->check_momentum_conservation(interaction);
}
```
But what I'm suggesting is that if you haven't done that, we could add it to this PR.

Comment (vrpascuzzi) [reply test/physics/em/PhotoelectricInteractor.test.cc:117]: Ah, gotcha! Makes sense.

I've done this but it's fine to add to this PR is fine -- it's a simple change.

Comment (amandalund) [reply test/physics/em/PhotoelectricInteractor.test.cc:117]: Thanks @vrpascuzzi! I'll go ahead and add it since I need it for the Livermore interactor.

Comment (sethrj) [reply test/physics/em/PhotoelectricInteractor.test.cc:117]: I'll leave that to you and @amandalund -- I'm OK with merging this branch, and then the BetheHeitler, and then cleaning up and checking whatever conservation we can at that point.

**Comment by sethrj** [comment]
@amandalund we can skip the extra conservation checking, but could you add energy deposition gold values to make sure they're sensible? And it looks like you may need to rebase to get rid of the conflict (which is surely my fault, sorry!).

File: test/physics/em/PhotoelectricInteractor.test.cc
Comment (amandalund) [reply test/physics/em/PhotoelectricInteractor.test.cc:117]: Ok that works!

**Review by sethrj — APPROVED** [review]

**Pull request merged**
